source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1139 +25,3368 @@
+ * send keep alive messages on connected but dormant sockets. Set
+ * ipconfigTCP_KEEP_ALIVE to 0 to prevent the automatic transmission of
+ * keep alive messages.
+ *
+ * If FreeRTOS-Plus-TCP does not receive a reply to a keep alive message
+ * then the connection will be broken and the socket will be marked as
+ * closed. Subsequent FreeRTOS_recv() calls on the socket will return
+ * -pdFREERTOS_ERRNO_ENOTCONN.
+ */
+
+#ifndef ipconfigTCP_KEEP_ALIVE
+    #define ipconfigTCP_KEEP_ALIVE    ipconfigDISABLE
+#endif
+
+#if ( ( ipconfigTCP_KEEP_ALIVE != ipconfigDISABLE ) && ( ipconfigTCP_KEEP_ALIVE != ipconfigENABLE ) )
+    #error Invalid ipconfigTCP_KEEP_ALIVE configuration
+#endif
+
+/*-----------------------------------------------------------------------*/
+
+/*#if ( ipconfigIS_ENABLED( ipconfigTCP_KEEP_ALIVE ) )*/
+
+/*-------------------------------------------------------------------*/
+
+/*
+ * ipconfigTCP_KEEP_ALIVE_INTERVAL
+ *
+ * https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_IP_Configuration.html#ipconfigTCP_KEEP_ALIVE_INTERVAL
+ *
+ * Type: TickType_t
+ * Unit: Seconds
+ * Minimum: 0
+ * Maximum: portMAX_DELAY / configTICK_RATE_HZ
+ *
+ * If ipconfigTCP_KEEP_ALIVE is set to 1 then
+ * ipconfigTCP_KEEP_ALIVE_INTERVAL sets the interval in seconds between
+ * successive keep alive messages. Keep alive messages are not sent at
+ * all unless ipconfigTCP_KEEP_ALIVE_INTERVAL seconds have passed since
+ * the last packet was sent or received.
+ */
+
+#ifndef ipconfigTCP_KEEP_ALIVE_INTERVAL
+    #define ipconfigTCP_KEEP_ALIVE_INTERVAL    ( 20 )
+#endif
+
+#if ( ipconfigTCP_KEEP_ALIVE_INTERVAL < 0 )
+    #error ipconfigTCP_KEEP_ALIVE_INTERVAL must be at least 0
+#endif
+
+#if ( ipconfigTCP_KEEP_ALIVE_INTERVAL > SIZE_MAX )
+    #error ipconfigTCP_KEEP_ALIVE_INTERVAL must be at most portMAX_DELAY / configTICK_RATE_HZ
+#endif
+
+/*-------------------------------------------------------------------*/
+
+/*#endif*/ /* if ( ipconfigIS_ENABLED( ipconfigTCP_KEEP_ALIVE ) ) */
+
+/*-----------------------------------------------------------------------*/
+
+/*
+ * ipconfigTCP_MSS
+ *
+ * https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_IP_Configuration.html#ipconfigTCP_MSS
+ *
+ * Type: size_t
+ * Unit: bytes
+ * Minimum: 536 ( tcpMINIMUM_SEGMENT_LENGTH )
+ *
+ * Sets the MSS value (in bytes) for all TCP packets.
+ *
+ * Note that FreeRTOS-Plus-TCP contains checks that the defined
+ * ipconfigNETWORK_MTU and ipconfigTCP_MSS values are consistent with each
+ * other.
+ *
+ * The default definition of 'ipconfigTCP_MSS' works well for most projects.
+ * Note that IPv6 headers are larger than IPv4 headers, leaving less space for
+ * the TCP payload. In prvSocketSetMSS_IPV6(), 20 bytes will be subtracted from
+ * 'ipconfigTCP_MSS'.
+ *
+ * The default is derived from MTU - ( ipconfigNETWORK_MTU + ipSIZE_OF_TCP_HEADER )
+ */
+
+#ifndef ipconfigTCP_MSS
+    #define ipconfigTCP_MSS    ( ipconfigNETWORK_MTU - 40 )
```

Comment:
Can we add it like 40 as this ( ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_TCP_HEADER ) )
As here it is not clear where 40 is coming from